coins/ethereum/contracts/Schnorr.sol

```diff
@@ -0,0 +1,18 @@
+//SPDX-License-Identifier: LGPLv3
+pragma solidity ^0.8.0;
+
+// see https://github.com/noot/schnorr-verify for implementation details
+contract Schnorr {
+    // sr := -s*P_x
+    // er := -e*P_x
+    // px := public key x-coord
+    // parity := public key y-coord parity (27 or 28)
+    // message := 32-byte message
+    // e := schnorr signature challenge
+    function verify(bytes32 sr, bytes32 er, bytes32 px, uint8 parity, bytes32 message, bytes32 e) public view returns (bool) {
+        // ecrecover = (m, v, r, s);
+        require(sr != 0);
+        address q = ecrecover(sr, parity, px, er);
+        return e == keccak256(abi.encodePacked(q, uint8(parity), px, block.chainid, message));
```

Comment:
While I'm unsure about which 0 checks are needed and which are offered, q should equal px doesn't have to be double transcripted here.

er is not bound to e and accordingly, this is trivially forgeable.

Reply:
right, added the preprocessing to the contract, it now accepts schnorr sig `(s, e)` instead.